pkg/factory/virtualservice/virtualservice.go

```diff
@@ -168,28 +171,66 @@ func (f *VirtualServiceFactory) AccessLog(ctx context.Context) (*accesslogv3.Acc
 		return nil, nil
 	}
 
+	if f.Spec.AccessLog != nil && f.Spec.AccessLogConfig != nil {
+		return nil, errors.New(errors.MultipleAccessLogConfigMessage)
+	}
+
 	if f.Spec.AccessLog != nil {
 		data = f.Spec.AccessLog.Raw
+
+		if err := options.Unmarshaler.Unmarshal(data, &accessLog); err != nil {
+			return nil, errors.WrapUKS(err, errors.UnmarshalMessage)
+		}
+
+		if err := accessLog.ValidateAll(); err != nil {
+			return nil, errors.WrapUKS(err, errors.CannotValidateCacheResourceMessage)
+		}
 	}
 
 	if f.Spec.AccessLogConfig != nil {
-		if f.Spec.AccessLog != nil {
-			return nil, errors.New(errors.MultipleAccessLogConfigMessage)
-		}
 		accessLogConfig := &v1alpha1.AccessLogConfig{}
 		err := f.client.Get(ctx, f.Spec.AccessLogConfig.NamespacedName(f.Namespace), accessLogConfig)
 		if err != nil {
 			return nil, errors.Wrap(err, errors.GetFromKubernetesMessage)
 		}
 		data = accessLogConfig.Spec.Raw
-	}
 
-	if err := options.Unmarshaler.Unmarshal(data, &accessLog); err != nil {
-		return nil, errors.WrapUKS(err, errors.UnmarshalMessage)
-	}
+		if err := options.Unmarshaler.Unmarshal(data, &accessLog); err != nil {
+			return nil, errors.WrapUKS(err, errors.UnmarshalMessage)
+		}
 
-	if err := accessLog.ValidateAll(); err != nil {
-		return nil, errors.WrapUKS(err, errors.CannotValidateCacheResourceMessage)
+		if val, ok := accessLogConfig.Annotations[options.AutoGeneratedFilename]; ok {
```

Comment:
copy-paste, it would be better to move the logic with annotation checking into a separate method